fw/sock_clnt.c

```diff
@@ -620,28 +624,105 @@ tfw_sock_clnt_cfgend(void)
 	return 0;
 }
 
+static int
+tfw_listen_socks_array_cmp(const void *l, const void *r)
+{
+	TfwAddr *a = &(*(TfwListenSock **)l)->addr;
+	TfwAddr *b = &(*(TfwListenSock **)r)->addr;
+	int cmp = memcmp(&a->sin6_addr, &b->sin6_addr, sizeof(a->sin6_addr));
+
+	if (cmp)
+		return cmp;
+	else
+		return a->sin6_port - b->sin6_port;
```

Comment:
```suggestion
	return (int)a->sin6_port - (int)b->sin6_port;
```
`sin6_port` is unsigned short, so it seems that a negative result will be presented just as a 2 byte positive integer